applications/GeoMechanicsApplication/custom_processes/apply_component_table_process.hpp

```
@@ -79,13 +81,22 @@ class ApplyComponentTableProcess : public Process
     {
         KRATOS_TRY
 
-        const Variable<double> &var = KratosComponents< Variable<double> >::Get(mVariableName);
-
-        block_for_each(mrModelPart.Nodes(), [&var, this](Node& rNode) {
-            if (mIsFixed) rNode.Fix(var);
-            else if (mIsFixedProvided) rNode.Free(var);
-            rNode.FastGetSolutionStepValue(var) = mInitialValue;
-        });
+        const auto& var = KratosComponents<Variable<double>>::Get(mVariableName);
+
+        auto variable_name_1 = mpTable->NameOfX();
+        boost::to_upper(variable_name_1);
+        if (variable_name_1 == "TIME") {
+            block_for_each(mrModelPart.Nodes(), [&var, this](Node& rNode) {
+                if (mIsFixed) rNode.Fix(var);
+                else if (mIsFixedProvided) rNode.Free(var);
+                rNode.FastGetSolutionStepValue(var) = mInitialValue;
+            });
+        }
+        else if (variable_name_1 == "X") {
+            block_for_each(mrModelPart.Nodes(), [&var, this](auto& node){
+                node.FastGetSolutionStepValue(var) = mpTable->GetValue(node.X());
+            });
```

Are we sure we don't need to support also "Y" (or "Z", whatever the other direction is in the horizontal plane)? I know it's only one dimension at a time, but maybe Embankment does not always use the same axis?

---

I have only added what we need now. As far as I can see, for the Embank team only "X" will suffice (since their models assume a plane strain situation, with the gravity direction pointing downwards). It is rather straightforward to add the other ones (i.e. "Y" and "Z") on request. What we could do now is to print a message when the function has no effect instead of silently returning.

---

Okay, clear. I leave it up to you if you want to add the message/error if something else than TIME or X is used.

---

Done.